### ISSUE TITLE
Variation price in cart and checkout

### ISSUE THREAD

**Issue by JohannLesacher** [hyyan/woo-poly-integration#15]
Hi !

Thanks for your work man you saved me a lot of coding :)

I've noticed an issue about the price displayed in the cart and in the review order table in the checkout page.
The price on the cart page is the price of the cheapest variation of the displayed product.
The price on the checkout page is 0.

I have :
Hyyan WooCommerce Polylang Integration 0.25
WooCommerce 2.4.1
Polylang 1.7.12

Without touching anything else, the desactivation of Hyyan WooCommerce Polylang Intergration resolves the problem so I guessed it has something to do with it.

Let me know if you want any other information but I'm pretty sure you are already aware of the problem. If I can help feel free to ask.

Thanks for your otherwise very great job, I think it's a great plugin.

Best regards,

Johann

**Comment by ibes** [comment]
I have the same problem.

I guess this line let woocommerce "forget" about the variation:
https://github.com/hyyan/woo-poly-integration/blob/master/src/Hyyan/WPI/Cart.php#L93

There has to be a check, if the Product is a Variable Product and if so, use the variation_id instead of product_id.

I guess something like this could be a starting point:
`$id = $cartItem['variation_id'] ? $cartItem['variation_id'] : $cartItem['product_id'];

        $translation = Utilities::getProductTranslationByID(
                        $id
        );
`

But this doesn't solve it.

I don't yet get, what else has to be changed get the necessary information.

Maybe the `woocommerce_cart_item_product_id` hook would also be needed.

Woocommerce seems to be quite complex or at least I don't get the magic that is happening ;)

**Comment by pmualaba** [comment]
Same problem over here. There is a way to TEMPORARILY (soft)fix it when you do the following:

1. To fix "incorrect prices in Cart and Checkout":
delete following code from /plugins/woo-poly-integration/src/Hyyan/WPI/cart.php 
// handle the translation of displayed porducts in cart
        add_filter(
                'woocommerce_cart_item_product'
                , array($this, 'translateCartProducts')
                , 10
                , 2
        );

2. To fix "not updating prices of translated variable products when you select another variation":
- Uncheck Fields Locker from WooPoly settings page (Features Tab) .
- now open the translated version of each variable product. Set the variation attributes manually and click the Save Changes button (Ajax). Now it works....until you click the UPDATE (Post) button. This erases the changes you just made and you have to go set all variation attributes again, and again only use the Save Changes button. 
Of course this is not a workable solution, but may help to troubeshoot the issue...

**Comment by Couiljer** [comment]
Hi guys, Did anyone found a solution to this cart problem? :confused:

**Comment by kochhase** [comment]
maybe this
https://gist.github.com/mister/2e58e64a2c43b5a130d9/revisions

**Comment by firefox747** [comment]
Hi everybody,
i found a fix for me.
I could not find a way to get the language of the product variation directly. Therefore i load every variation and check against the current language. Not sure if it is the most elegant way, but it works for me.

    public function translateCartProducts($cartItemData, $cartItem)
    {
        $translation = false;
        // Try to translate the product variation
        if (isset($cartItem['variation_id'])) {
            // Get variations of current cart item
            $variations = Product\Variation::getRelatedVariation($cartItem['variation_id'], true);
            foreach ($variations as $variation) {
                // Load the product
                $product = \wc_get_product($variation);
                // Check for correct language
                if (pll_get_post_language($product->id) == pll_current_language()) {
                    $translation = $product;
                }
            }
        }
        // Fallback to product translation
        if (!$translation) {
            $translation = Utilities::getProductTranslationByID($cartItem['product_id']);
        }
        // Return untranslated cart item in case no translation was found.
        return $translation ? $translation : $cartItemData;
    }

**Comment by Couiljer** [comment]
I tried Firefox747 proposal but doesn't solve my problem. The translated cart is not working anymore then. And wrong price still apear on standard page. Thx anyway.

**Comment by Couiljer** [comment]
@kochhase Thanks to you ! I replaced the plugins php with the one you linked. It seems to work nicely.

**Comment by dkuster** [comment]
Hi everybody! Nearly 6 months ago I followed @kochhase link to fix the issue and forgot about it. 

Yesterday I upgraded to:
- woocommerce 2.6.4
- polylang 2.0
- Hyyan WooCommerce Polylang Integration 0.27

And of course the issue came back when I applied the new release :(

Any chance that this fix will be integrated in next version?

**Comment by pierrese** [comment]
Hello,

Same problem, solved thank to that : https://wordpress.org/support/topic/variable-product-price-on-cart-page-is-wrong?replies=7

**Comment by nicmare** [comment]
i found a reliable workaround for this. you will need a helper script: https://github.com/herewithme/wp-filters-extras/
then put this in your functions.php:
```php
function remove_registration_fields(){
    require_once("wp-filters-extras.php");
    remove_filters_with_method_name("woocommerce_cart_item_product","translateCartProducts",10);
}
add_action( 'wp', 'remove_registration_fields' );
```

**Comment by Jon007** [comment]
this was fixed by decarvalhoaa in https://github.com/hyyan/woo-poly-integration/pull/64 and released in version 0.28  
.. and is still working in version 1.0
